pkg/env/env.go

```diff
@@ -0,0 +1,175 @@
+package env
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"strconv"
+	"time"
+)
+
+// ErrUnsupportedType is returned when an environment variable target to bind is
+// not supported. For example a custom struct type.
+var ErrUnsupportedType = errors.New("unsupported type")
+
+// ErrNotAPointer is returned when an environment variable target to bind is not
+// a pointer. It's also a wrapped "unsupported type" error.
+var ErrNotAPointer = fmt.Errorf("%w: not a pointer", ErrUnsupportedType)
+
+// ErrParse is used in the ParseError type when checking error.Is to be able to
+// identify the error responses from the bind functions.
+var ErrParse = errors.New("failed to parse")
+
+// ParseError is an error type that unwraps to the internal parsing error
+// obtained.
+type ParseError struct {
+	EnvKey   string
+	EnvValue string
+	Err      error
+}
+
+// Error returns the error string. Makes it compliant with the error interface.
+func (err ParseError) Error() string {
+	return fmt.Sprintf("env: %q: %q: %s", err.EnvKey, err.EnvValue, err.Err)
+}
+
+// Is returns true if the target error is ErrParse or if the inner actual
+// parsing error is the same error.
+//
+// This method provides compatability with the errors.Is function.
+func (err ParseError) Is(target error) bool {
+	return errors.Is(ErrParse, target) || errors.Is(err.Err, target)
+}
+
+// As returns true if this error could be unwrapped into the type of the target
+// error.
+//
+// This method provides compatability with the errors.As function.
+func (err ParseError) As(target interface{}) bool {
+	return errors.As(err.Err, target)
+}
+
+// Unwrap returns the inner actual parsing error, such as the error return value
+// from a strconv function.
+//
+// This method provides compatability with the errors.Unwrap function.
+func (err ParseError) Unwrap() error {
+	return err.Err
+}
+
+// Bind will take a value pointer and depending on its type will try to parse
+// the environment variable, if set and not empty, using the appropriate parsing
+// function.
+//
+// Not all types are supported, and so except for parsing errors, you may also
+// get a wrapped env.ErrUnsupportedType error returned. If the target interface
+// is not a pointer, a wrapped env.ErrNotAPointer error is returned.
+//
+// Parsing errors are wrapped inside the env.ParseError type.
```

Comment:
Go docs does works a little like markdown in the sense that lines after each other are just joined together. But your suggestion does provide a new kind of symmetry in the comment.

So your comment would be equivalent to:

```suggestion
// Returns an env.ParseError on parsing errors. Returns a wrapped
// env.ErrUnsupportedType error if the type of the interface is not supported.
// Returns a wrapped env.ErrNotAPointer error if the target interface is not a
// pointer. Returns nil otherwise.
```

Or if you want to enforce one per line, even over at https://pkg.go.dev/github.com/iver-wharf/wharf-core, then something like this would be better suited:

```suggestion
// Returns an env.ParseError on parsing errors.
//
// Returns a wrapped env.ErrUnsupportedType error if the type of the interface
// is not supported.
//
// Returns a wrapped env.ErrNotAPointer error if the target interface is not a
// pointer.
//
// Returns nil otherwise.
```

What do you think?

Reply:
Although I definitely think it would be nicer in the docs if it was one per line, it does feel a bit bloated.
Most IDEs can hide blocks of comments though, so that might be okay... Not sure tbh

Reply:
The `Bind` is the most important function in this package, so I think it's fine if the docs are a bit bloated.